concent_api/core/views.py

```diff
@@ -1443,6 +1443,114 @@ def store_message(
     return stored_message
 
 
+def update_concent_states(
+    client_public_key: str,
+):
+    active_subtask_states = [
+        'forcing_report',
+        'forcing_result_transfer',
+        'forcing_acceptance',
+        'additional_verification',
+    ]
+    clients_subtask_list_as_requestor   = Subtask.objects.filter(requestor = client_public_key, delivered = False, state__in = active_subtask_states)
+    clients_subtask_list_as_provider    = Subtask.objects.filter(provider = client_public_key,  delivered = False, state__in = active_subtask_states)
+
+    if len(clients_subtask_list_as_requestor) == 0 and len(clients_subtask_list_as_provider) == 0:
+        logging.log_no_changes_in_subtask_states(client_public_key)
+    else:
+        if len(clients_subtask_list_as_requestor) != 0:
+            update_states_of_subtasks(clients_subtask_list_as_requestor)
+            logging.log_changes_in_subtask_states(
+                client_public_key,
+                len(clients_subtask_list_as_requestor),
+                'requestor',
+            )
+        if len(clients_subtask_list_as_provider) != 0:
+            update_states_of_subtasks(clients_subtask_list_as_provider)
+            logging.log_changes_in_subtask_states(
+                client_public_key,
+                len(clients_subtask_list_as_provider),
+                'provider',
+            )
+
+def update_states_of_subtasks(
+    subtask_list:       list,
+    client_public_key:  str,
+):
+    current_time = get_current_utc_timestamp()
+
+    for subtask in subtask_list:
+        receive_queue_item = ReceiveQueue.objects.get(
+            client  = client_public_key,
+            subtask = subtask,
+        )
+        if subtask.next_deadline.timestamp() <= current_time:
+            if subtask.state == Subtask.SubtaskState.FORCING_REPORT:
```

Comment:
This part is not what it should be. Here you should create entries for `SubtaskResultsSettled`, `VerdictReportComputedTask` etc. Please go through our current views and examine cases when we check timeouts. We can discuss it tomorrow.